### PR TITLE
Add ZSTD Compression Option

### PR DESCRIPTION
**Pull request by evolvedmicrobe**
Allows us to experiment with different compressors, modeled after `lz4_file`

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1009,29 +1028,29 @@ checksum = "bebd363326d05ec3e2f532ab7660680f3b02130d780c299bca73469d521bc0ed"
...
-version = "1.0.156"
+version = "1.0.160"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "d7e29c4601e36bcec74a223228dce795f4cd3616341a4af93520ca1a837c087d"
+checksum = "291a097c63d8497e00160b166a967a4a79c64f3facdd01cbd7502231688d77df"
 dependencies = [
  "proc-macro2",
  "quote",
- "syn",
+ "syn 2.0.15",
```
Comment (sreenathkrishnan) [inline Cargo.lock:1046]: We are probably not ready for `syn 2`. It's also a fairly big dependency, so would be better to wait until we can completely move all of `syn 1.x` to `syn 2

**Review by sreenathkrishnan — APPROVED** [review]
Awesome!

**Review by macklin-10x — APPROVED** [review]
Nice!  Zstd is impressive and a good default for anything internal, this will definitely aid adoption.  Cargo.lock had a lot of churn including some dependency downgrades, it might be a good idea to try to clean that up before merging.  Given that this is used as a library, I'm honestly not clear on why Cargo.lock is checked into the repo in the first place...

**Comment by evolvedmicrobe** [comment]
> Nice! Zstd is impressive and a good default for anything internal, this will definitely aid adoption. Cargo.lock had a lot of churn including some dependency downgrades, it might be a good idea to try to clean that up before merging. Given that this is used as a library, I'm honestly not clear on why Cargo.lock is checked into the repo in the first place...

Yeah, why is it checked in?? And also, I am totally confused how I wound up changing so many seeming unrelated things in the `Cargo.lock` file here.

**Comment by sreenathkrishnan** [comment]
`Cargo.lock` is checked in so that we use consistent crate versions in the CI. The versions in there do not matter for extrernal crates.

**Comment by evolvedmicrobe** [comment]
> `Cargo.lock` is checked in so that we use consistent crate versions in the CI. The versions in there do not matter for extrernal crates.

That makes a lot of sense.  What doesn't make a lot of sense is how I apparently did not pull from master prior to starting on this PR, and that created a bunch of unnecessary churn, fixing that now...

**Comment by evolvedmicrobe** [comment]
And I bjorked this PR pretty badly - back in the day I didn't have access to this repo, so I forked it, didn't realize I was pulling from that outdated fork when "updating" master, and lots of unnecessary churn as a result.

Closing in favor of: https://github.com/martian-lang/martian-rust/pull/343 - time to start fresh

**Pull request closed without merging**